bundle/config/mutator/run_as.go

```diff
@@ -23,12 +27,120 @@ func (m *setRunAs) Name() string {
 	return "SetRunAs"
 }
 
+// Resources that satisfy one of the following conditions:
+//  1. Allow to set run_as for the resources to a different user from the current
+//     deployment user. For example, jobs.
+//  2. Does not make sense for these resources to run_as a different user. We do not
+//     have plans to add platform side support for `run_as` for these resources.
+//     For example, experiments or registered models.
+var allowListForRunAs = []string{"jobs", "models", "registered_models", "experiments"}
+
+// Resources that do not allow setting a run_as identity to a different user but
+// have plans to add platform side support for `run_as` for these resources at
+// some point in the future. For example, pipelines, model serving endpoints or lakeview dashboards.
+//
+// We expect the allow list and the deny list to form mutually exclusive and exhaustive
+// sets of all resource types that are supported by DABs.
+var denyListForRunAs = []string{"pipelines", "model_serving_endpoints"}
+
+type errorUnsupportedResourceTypeForRunAs struct {
+	resourceType  string
+	resourceValue dyn.Value
+	currentUser   string
+	runAsUser     string
+}
+
+// TODO(6 March 2024): This error message is big. We should split this once
+// diag.Diagnostics is ready.
+// TODO(6 March 2024): Link the docs page describing run_as semantics in the error below
+// once the page is ready.
+func (e errorUnsupportedResourceTypeForRunAs) Error() string {
+	return fmt.Sprintf("%s are not supported when the current deployment user is different from the bundle's run_as identity. Please deploy as the run_as identity. List of supported resources: [%s]. Location of the unsupported resource: %s. Current identity: %s. Run as identity: %s", e.resourceType, strings.Join(allowListForRunAs, ", "), e.resourceValue.Location(), e.currentUser, e.runAsUser)
+}
```

Comment:
Is the "list of supported resources" actually relevant in this error?

Reply:
> Do you want to keep this a todo until you have a doc page to link to?
Yeah lets keep the TODO in code. I don't think this warrants a ticket in our backlog.

> Is the "list of supported resources" actually relevant in this error?
The original intention was to avoid users running into this error multiple times as they remove the resources one by one. Removed however to make the error more terse as you suggested.